src/templates.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2014 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_TEMPLATES_H
+#define BITCOIN_TEMPLATES_H
+
+#include <assert.h>
+#include <cstddef>
+
+template<typename T>
+class Container
```

Comment:
1)
Not sure, but we might prefer something like smart c++11 pointers (long term, once switched to c++11) over boost managed pointers (to – at least – not increase the boost library adhesion).

2)
If we stick with the `Container` class: `Container` as classname in the global space seems to be to general.